spring-web/src/main/java/org/springframework/http/HttpHeaders.java

```diff
@@ -1575,28 +1573,28 @@ public void clearContentHeaders() {
 	 */
 	protected List<String> getETagValuesAsList(String headerName) {
 		List<String> values = get(headerName);
-		if (values != null) {
-			List<String> result = new ArrayList<>();
-			for (String value : values) {
-				if (value != null) {
-					Matcher matcher = ETAG_HEADER_VALUE_PATTERN.matcher(value);
-					while (matcher.find()) {
-						if ("*".equals(matcher.group())) {
-							result.add(matcher.group());
-						}
-						else {
-							result.add(matcher.group(1));
-						}
+		if (values == null) {
+			return Collections.emptyList();
+		}
+		List<String> result = new ArrayList<>();
+		for (String value : values) {
+			if (value != null) {
```

Comment:
It would be good to `continue` if `value` is `null`.

Reply:
yes, indeed. Fixed